docs/cloud/pouta/api-access.md

```diff
@@ -0,0 +1,140 @@
+You coud make requests to the Pouta cloud. Normally you would start with 
+authenticating yourself as: 
+
+```
+curl -i   -H "Content-Type: application/json"   -d '
+{"auth": {
+    "identity": {
+      "methods": ["password"],
+      "password": {
+        "user": {
+          "name": "'$OS_USERNAME'",
+          "domain": {"name": "'$OS_USER_DOMAIN_NAME'"},
+          "password": "'$OS_PASSWORD'"
+        }
+      }
+    },
+    "scope": {
+      "project": {
+        "domain": {"id": "'$OS_PROJECT_DOMAIN_ID'"},
+        "name": "'$OS_PROJECT_NAME'"
+      }
+    }
+  }
+}' "$OS_AUTH_URL/auth/tokens?nocatalog" | python -m json.tool
+```
+
+And obtaining your token as an `X-Subject-Token` response header. The response 
+body for our request will also contain additional useful information which includes 
+the expiration date and time of the token as `"expires_at":"datetime"`.  
+
+Once authenticated, we can make further CRUD requests to the various APIs handling our 
+cloud resources. For example, we can request the compute API for the list of available 
+virtual machine flavors as:
+
+```
+export OS_TOKEN=<copy-your-token-here>
+export OS_COMPUTE_API=https://pouta.csc.fi:8777/v2.1
+```
+
+```
+curl -s -H "X-Auth-Token: $OS_TOKEN" \
+  $OS_COMPUTE_API/flavors \
+  | python -m json.tool
+```
+
+You can consult the [Pouta web interface](https://pouta.csc.fi/dashboard/project/api_access/) further 
+to get the right values for the Pouta API endpoints such as `OS_COMPUTE_API`.  
+
+### Pouta access through client libraries
+
+[Openstacksdk](https://docs.openstack.org/openstacksdk/latest/) is a client library (SDK) for 
+building applications and services that work with OpenStack Clouds. It provides a consistent and complete 
+set of features to interact with the various OpenStack components. The SDK implements Python 
+bindings to the OpenStack API, which enables you to perform automation tasks in Python by 
+making calls on Python objects rather than making REST calls directly.
+
+In order to use it with our applications we need to first install the SDK as: 
+
+`pip install openstacksdk`
+
+Next, we need to provide our configurations and credentials through a `clouds.yaml` file which can be downloaded from 
+the [Pouta web interface](https://pouta.csc.fi/dashboard/project/api_access/) after logging in. 
+Openstacksdk expects this file in one of the following folders: the `current` directory, the `~/.config/openstack` 
+directory, or `/etc/openstack` directory. For quick test purposes, you can add your password in the `clouds.yaml` as: 
+
+```
+clouds:
+  openstack:
+    auth:
+      auth_url: https://pouta.csc.fi:5001/v3
+      username: "username"
+      password: XXXXXXXXXXX
+      project_id: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+      project_name: "project_xxxxxx"
+      user_domain_name: "Default"
+    regions:
+    - regionOne
+    interface: "public"
+    identity_api_version: 3
+```
```

Comment:
I am not sure this example is faster than creating the `secure.yaml` file directly.

Reply:
I think it is good to show both options and let the user make the choice. The `secure.yaml` is also an optional so keeping the text as is makes that fact clear.

Reply:
I do not agree. What is the advantage of the first option? That users will be able to commit a file with a password by mistake?

Reply:
Ok. The password is now only under `secure.yaml`. Hopefully users do not commit that file by mistake too :).

Reply:
Thanks, they will probably do it anyways :smile: